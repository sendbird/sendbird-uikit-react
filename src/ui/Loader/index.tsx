import React, { ReactElement } from 'react';

import './index.scss';

import Icon, { IconTypes } from '../Icon';

export interface LoaderProps {
  className?: string | Array<string>;
  width?: string | number;
  height?: string | number;
  children?: ReactElement;
}

export default function Loader({
  className = '',
  width = '26px',
  height = '26px',
  children,
}: LoaderProps): ReactElement {
  return (
    <div
      className={[
        ...(Array.isArray(className) ? className : [className]),
        'sendbird-loader',
      ].join(' ')}
      style={{
        width: typeof width === 'string' ? width : `${width}px`,
        height: typeof height === 'string' ? height : `${height}px`,
      }}
    >
      {
        children
        || (
          <Icon
            type={IconTypes.SPINNER}
            width="26px"
            height="26px"
          />
        )
      }
    </div>
  );
}
