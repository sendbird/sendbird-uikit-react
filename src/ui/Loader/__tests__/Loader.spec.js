import React from 'react';
import { render } from '@testing-library/react';

import Loader from "../index";

describe('ui/Loader', () => {
  it('should do a snapshot test of the default Loader DOM', function () {
    const { asFragment } = render(
      <Loader />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
