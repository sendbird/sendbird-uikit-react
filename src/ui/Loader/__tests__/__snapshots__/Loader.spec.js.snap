// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader should do a snapshot test of the default Loader DOM 1`] = `
<div
  className=" sendbird-loader"
  style={
    Object {
      "height": "26px",
      "width": "26px",
    }
  }
>
  <div
    className=" sendbird-icon sendbird-icon-spinner "
    onClick={[Function]}
    onKeyDown={[Function]}
    role="button"
    style={
      Object {
        "height": "26px",
        "minHeight": "26px",
        "minWidth": "26px",
        "width": "26px",
      }
    }
    tabIndex="0"
  >
    <test-file-stub />
  </div>
</div>
`;
